bcgov/HOUS-permit-portal
# Track part 3 checklist completion status

## Description

Adds logic for tracking completion status and implements start page UI for POC

## What type of PR is this? (check all applicable)


## Related Tickets & Documents

[BPHH-2453](https://hous-bssb.atlassian.net/browse/BPHH-2453)
[BPHH-2441](https://hous-bssb.atlassian.net/browse/BPHH-2441)

## Steps to QA

1. part 3 step code is visible on public page (`/part-3-step-code`). Click CTA on start page to complete the step and navigate to the next section. Note - currently a new step code is created on every page reload (see TODO)
2. part 3 step code is visible on permit application page (`/permit-applications/:id/edit/part-3-step-code`). Click CTA on start page to complete the step and navigate to the next section. Section completion status should be persisted on page refresh. Upon completing the start page, navigate back to the base URL (`/permit-applications/:id/edit/part-3-step-code`) and verify the user is redirected to the first incomplete step (e.g. instead of to the start page).

## UI Accessibility Checklist



_Note the following checklist is not an exhaustive list. Check
the [WebAIM checklist](https://webaim.org/standards/wcag/checklist) for
a user-friendly checklist with more details. Aim to meet at least *AA* conformance level where applicable. Check
the [WCAG guidelines](https://webaim.org/standards/wcag/checklist) for the full guidelines._

      with [Wave Browser Extension](https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh)
      for accessibility errors?

## General Checklist


## [optional] Are there any post-deployment tasks we need to perform?

n/a


[BPHH-2453]: https://hous-bssb.atlassian.net/browse/BPHH-2453?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[BPHH-2441]: https://hous-bssb.atlassian.net/browse/BPHH-2441?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
